azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/parser/ConfigParser.java

```diff
@@ -0,0 +1,128 @@
+import com.microsoft.azure.maven.containerapps.config.DeploymentType;
+import com.microsoft.azure.maven.containerapps.config.IngressMavenConfig;
+import com.microsoft.azure.maven.utils.MavenUtils;
+import com.microsoft.azure.toolkit.lib.Azure;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.utils.Utils;
+import com.microsoft.azure.toolkit.lib.containerapps.config.ContainerAppConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.config.ContainerAppsEnvironmentConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.containerapp.ContainerAppDraft;
+import com.microsoft.azure.toolkit.lib.containerapps.model.IngressConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.model.ResourceConfiguration;
+import com.microsoft.azure.toolkit.lib.containerregistry.AzureContainerRegistry;
+import com.microsoft.azure.toolkit.lib.containerregistry.ContainerRegistry;
+import com.microsoft.azure.toolkit.lib.containerregistry.config.ContainerRegistryConfig;
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class ConfigParser {
+
+    protected AbstractMojoBase mojo;
+
+    private final String timestamp;
+
+    public ConfigParser(AbstractMojoBase mojo) {
+        this.mojo = mojo;
+        this.timestamp = Utils.getTimestamp();
+    }
+
+    public ContainerAppConfig getContainerAppConfig() {
+        final ContainerAppConfig config = new ContainerAppConfig();
+        final ContainerAppsEnvironmentConfig envConfig = new ContainerAppsEnvironmentConfig();
+        envConfig.setSubscriptionId(mojo.getSubscriptionId());
+        envConfig.setResourceGroup(mojo.getResourceGroup());
+        envConfig.setAppEnvironmentName(mojo.getAppEnvironmentName());
+        envConfig.setRegion(mojo.getRegion());
+        config.setEnvironment(envConfig);
+        config.setAppName(mojo.getAppName());
+        config.setResourceConfiguration(getResourceConfigurationFromContainers(mojo.getContainers()));
+        config.setIngressConfig(getIngressConfig(mojo.getIngress()));
+        config.setRegistryConfig(getRegistryConfig());
+        config.setImageConfig(getImageConfigFromContainers(config));
+        config.setScaleConfig(mojo.getScale());
+        return config;
+    }
+
+    public ContainerAppDraft.ImageConfig getImageConfigFromContainers(ContainerAppConfig config) {
+        List<AppContainerMavenConfig> containers = mojo.getContainers();
+        if (containers == null || containers.isEmpty()) {
+            return null;
+        }
+        final String defaultImageName = String.format("%s%s/%s:%s", config.getRegistryConfig().getRegistryName(), ContainerRegistry.ACR_IMAGE_SUFFIX, mojo.getAppName(), timestamp);
+        final String fullImageName = Optional.ofNullable(containers.get(0).getImage()).orElse(defaultImageName);
```

Comment:
It seems that we only accept the first container configuration, if so, why not change mojo.getContainers to `AppContainerMavenConfig` instead of a List?

Reply:
Here we will spike if we should support multi container payload later. For the first edition, we can just keep the list for future scalability